include/pthread.h

```diff
@@ -344,9 +344,16 @@ typedef struct pthread_mutex_s pthread_mutex_t;
 #  define PTHREAD_MUTEX_INITIALIZER {NULL, SEM_INITIALIZER(1), -1, \
                                      __PTHREAD_MUTEX_DEFAULT_FLAGS, \
                                      PTHREAD_MUTEX_DEFAULT, 0}
+#  define PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP \
+                                     {NULL, SEM_INITIALIZER(1), -1, \
+                                     __PTHREAD_MUTEX_DEFAULT_FLAGS, \
+                                     PTHREAD_MUTEX_RECURSIVE, 0}
 #elif defined(CONFIG_PTHREAD_MUTEX_TYPES)
 #  define PTHREAD_MUTEX_INITIALIZER {SEM_INITIALIZER(1), -1, \
                                      PTHREAD_MUTEX_DEFAULT, 0}
+#  define PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP \
```

Comment:
Why 2 Times? What is NP?  How about a comment.  // nonportable

Reply:
>> Why 2 Times? 
thread safe and unsafe version

>> What is NP? How about a comment. // nonportable
Ok

Reply:
I added some comment of _NP in the commit message and Summary.

Reply:
Thanks @anchao - imissed the #elsif :)  I realize this is not part of the this change....but I wonder if the #if defing should nested not compound for better readability.